builtin/logical/pki/config_util.go

```diff
@@ -44,9 +45,17 @@ func (sc *storageContext) updateDefaultIssuerId(id issuerID) error {
 	}
 
 	if config.DefaultIssuerId != id {
-		return sc.setIssuersConfig(&issuerConfigEntry{
+		err := sc.setIssuersConfig(&issuerConfigEntry{
 			DefaultIssuerId: id,
 		})
+
+		if err != nil {
+			return err
+		}
+
+		// Can we assume that this fetch won't raise an error?
+		issuer, _ := sc.fetchIssuerById(id)
+		issuer.LastModified = time.Now().In(time.FixedZone("GMT", 0))
```

Comment:
I am not too sure but I don't think so.

Reply:
I don't think so, I have run all my manual tests without it and everything worked. Is what's commented below what you mean with the question?

Reply:
I might be missing something .... but `updateDefaultIssuerId` is called:

 1. During `config/issuer` operations, where its the last call (`path_config_ca.go`). So `issuer.LastModified` assignment without a `writeIssuer` call in this case will result in the modification not being persisted.
 2. When there's a new key (via `importKey`) and we lack a default issuer but have an issuer using this key. Then we'll call this, but again, there's no subsequent call to `writeIssuer`.
 3. Same holds for `importIssuer`.

However, these are kinda moot as the issue is locally scoped: the issuer is fetched in this function but not returned from this function, so the modifications will disappear at the end of the function return if they aren't persisted?

If you use `/sys/raw` [to read the stored issuer config](https://www.vaultproject.io/api-docs/system/raw) I think you'll see its not being persisted.

I'd imagine that the following would be enough to verify this:

1. Create two issuers, A and B.
2. Assign default to A.
3. Fetch with the header via `/cert/ca`. Use the current timestamp.
4. Wait a bit.
5. Set the default to B (`/config/issuers`).
6. Wait a bit.
7. Fetch with the header via `/cert/ca`. Use the timestamp of the last fetch, not right now. 

I think 7 will yield A, vs. B. Because while the issuer was modified since there, this write isn't issued. My 2c.?

Reply:
You are right, if `writeIssuer` isn't called here, when we change the default ref the issuer LastModified field isn't updated.

Reply:
Yes, you are right. If we don't writeIssuer the LastModified update isn't persisted.

Reply:
Cool, ty!